packages/bitcore-node/src/models/transaction.ts

```diff
@@ -519,35 +519,40 @@ export class TransactionModel extends BaseModel<ITransaction> {
     if (!initialSyncComplete || !spendOps.length) {
       return;
     }
-    let prunedTxs = new Set();
-    for (const spendOp of spendOps) {
-      let coin = await CoinStorage.collection.findOne(
-        {
-          chain,
-          network,
-          spentHeight: SpentHeightIndicators.pending,
-          mintTxid: spendOp.updateOne.filter.mintTxid,
-          mintIndex: spendOp.updateOne.filter.mintIndex,
-          spentTxid: { $ne: spendOp.updateOne.update.$set.spentTxid }
-        },
-        { projection: { spentTxid: 1 } }
-      );
-      if (coin && !prunedTxs.has(coin.spentTxid)) {
-        prunedTxs.add(coin.spentTxid);
-        await Promise.all([
-          this.collection.update(
-            { chain, network, txid: coin.spentTxid },
-            { $set: { blockHeight: SpentHeightIndicators.conflicting } },
-            { multi: true }
-          ),
-          CoinStorage.collection.update(
-            { chain, network, mintTxid: coin.spentTxid },
-            { $set: { mintHeight: SpentHeightIndicators.conflicting } },
-            { multi: true }
-          )
-        ]);
-      }
-    }
+    let coins = await CoinStorage.collection
+      .find({
+        chain,
+        network,
+        spentHeight: SpentHeightIndicators.pending,
+        mintTxid: { $in: spendOps.map(s => s.updateOne.filter.mintTxid) }
+      })
+      .project({ mintTxid: 1, mintIndex: 1, spentTxid: 1 })
+      .toArray();
+    coins = coins.filter(
+      c =>
+        spendOps.findIndex(
+          s =>
+            s.updateOne.filter.mintTxid === c.mintTxid &&
+            s.updateOne.filter.mintIndex === c.mintIndex &&
+            s.updateOne.update.$set.spentTxid !== c.spentTxid
+        ) > -1
+    );
+
+    const spentTxids = Array.from(new Set(coins.map(c => c.spentTxid)));
```

Comment:
a name indicating these will be conflicting would be more clear, spentTxids is very innocuous